### PR TITLE
chore: sort calendar crendentials in event manager

### LINKED ISSUE #11671: [CAL-2569] Fallback of using the first connected calendar isn’t reliable
As [we don’t explicitly sort the credentials in ascending order of ](https://github.com/calcom/cal.com/blob/1f6e80558b97f1b61d7c1885661611f2e0d57633/packages/core/EventManager.ts#L98)`[credentialId](https://github.com/calcom/cal.com/blob/1f6e80558b97f1b61d7c1885661611f2e0d57633/packages/core/EventManager.ts#L98)`, we get the list of credentials in descending order of last updated time in DB(so it seems) and thus whenever a new credential is added or existing credential is modified, we erroneously consider that credential as the first one, instead of the first added one actually.

This is problematic because depending on the calendar apps installation/uninstallation/updating, the target destination calendar might change without user knowing.

We should [sort like videoCredentials](https://github.com/calcom/cal.com/blob/1f6e80558b97f1b61d7c1885661611f2e0d57633/packages/core/EventManager.ts#L107)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2569](https://linear.app/calcom/issue/CAL-2569/fallback-of-using-the-first-connected-calendar-isnt-reliable)</sub>

### PR DESCRIPTION
**Pull request by Shaik-Sirajuddin**
## What does this PR do?

- Fixes #11671
- Fixes CAL-2569

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

**Comment by supalarry** [comment]
@hariombalhara I am confused - in the linked issue #11671 you wrote:
```
we erroneously consider that credential as the first one, instead of the first added one actually.
```

but by sorting it by DESC using:
```
.sort((a, b) => {
        return b.id - a.id;
      });
```
we have last added calendar first as the default fallback one, but the function for falling back is called `fallbackToFirstConnectedCalendar`.

So - do we want by default to fallback to the first connected calendar, or to the latest one?

If first connected:
1. We need to sort in ASCENDING order by id.
2. If by latest one, we need to rename `fallbackToFirstConnectedCalendar` to `fallbackToLastConnectedCalendar`.

**Review by supalarry — COMMENTED** [review]
@hariombalhara I am confused - in the linked issue https://github.com/calcom/cal.com/issues/11671 you wrote:
```
we erroneously consider that credential as the first one, instead of the first added one actually.
```

but by sorting it by DESC using:
```
.sort((a, b) => {
    return b.id - a.id;
});
```
we have last added calendar as the default fallback one, but the function for falling back is called `fallbackToFirstConnectedCalendar`.

So - do we want by default to fallback to the first connected calendar, or to the latest one?

If first connected, we need to sort in ASCENDING order by id. I think this makes most sense - fallback to first connected.

If by latest one, we need to rename `fallbackToFirstConnectedCalendar` to `fallbackToLastConnectedCalendar`.

**Comment by hariombalhara** [comment]
@supalarry Yeah the statement isn't correct. I meant we need to have the latest added one consistently which would be the first one if we do sort in descending.

The same way we do for `videoCredentials`
image

File: packages/core/EventManager.ts [hunk 1]
```diff
@@ -98,10 +98,15 @@ export default class EventManager {
...
-    );
+    this.calendarCredentials = appCredentials
+      .filter(
+        // Backwards compatibility until CRM manager is implemented
+        (cred) => cred.type.endsWith("_calendar") && !cred.type.includes("other_calendar")
+      )
+      //see https://github.com/calcom/cal.com/issues/11671#issue-1923600672
+      .sort((a, b) => {
+        return b.id - a.id;
+      });
```
Comment (hariombalhara) [inline packages/core/EventManager.ts:109]: how about we name this arrow function, `latestFirst` and reuse it  below as well. 
Makes the code more readable

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thanks @Shaik-Sirajuddin !! 
I left a quick to implement suggestion otherwise it's good.

File: packages/core/EventManager.ts (hunk 1 above)
Comment (Udit-takkar) [reply packages/core/EventManager.ts:109]: Do you mean `oldestFirst` as we are sorting in ascending order

**Review by Udit-takkar — DISMISSED** [review]

**Comment by Shaik-Sirajuddin** [comment]
> Thanks @Shaik-Sirajuddin !! I left a quick to implement suggestion otherwise it's good.

Updated function names as suggested
PTAL @hariombalhara

File: packages/core/EventManager.ts (hunk 1 above)
Comment (hariombalhara) [reply packages/core/EventManager.ts]: I think we are sorting in descending order only, unless I am misunderstanding

image

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**